core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -378,6 +378,28 @@ public void setRequestTimeout(Duration requestTimeout) {
         setRequestTimeoutMillis(requireNonNull(requestTimeout, "requestTimeout").toMillis());
     }
 
+    @Override
+    public void setRequestTimeoutAfterNanos(long requestTimeoutAfterNanos) {
+        if (requestTimeoutAfterNanos < 0) {
+            throw new IllegalArgumentException(
+                    "requestTimeoutAfterNanos: " + requestTimeoutAfterNanos + " (expected: >= 0)");
+        }
+
+        final RequestTimeoutController requestTimeoutController = this.requestTimeoutController;
+        if (requestTimeoutController != null) {
+            if (ch.eventLoop().inEventLoop()) {
+                requestTimeoutController.setDeadline(requestTimeoutAfterNanos);
```

Comment:
You must update `this.requestTimeoutMillis` whenever the effective timeout changes.

Reply:
I intended `requestTimeoutAfterNanos` reset the request timeout from now. It means

* request timeout : 100ms
* start time : 0000ms
* current time elapsed : 50ms
* requestTimeoutAfter : 100ms (from now)
* new request timeout : 100 - 50 + 100 = 150ms

But it is hard to know start time here. Without using the start time, it would be better to give up reset request timeout from current time. And we could simple add a API for extending timeout. Because the `this.requestTimeoutMillis` must be updated correctly.

Reply:
After reading this review (https://github.com/line/armeria/pull/2343#discussion_r361444153),  it is real deadline. `this.requestTimeoutMillis` could be updated by the specified deadline.

Reply:
> Because the this.requestTimeoutMillis must be updated correctly.

You can change the controller to return the new timeout.

Reply:
> You can change the controller to return the new timeout.

I thought the solution too, but I was not sure.😅 Thanks for clarifying.🙏